src/mlpack/methods/logistic_regression/logistic_regression_main.cpp

```diff
@@ -327,6 +337,21 @@ void BINDING_FUNCTION(util::Params& params, util::Timers& timers)
     }
   }
 
+  // Did we want training accuracy?
+  if (params.Has("print_training_accuracy"))
+  {
+    timers.Start("lr_prediction");
+    arma::Row<size_t> predictions;
+    model->Classify(regressors, predictions);
+
+    const size_t correct = arma::accu(predictions == responses);
+
+    Log::Info << correct << " of " << responses.n_elem << " correct on training"
+        << " set (" << (double(correct) / double(responses.n_elem) * 100) << ")."
+        << endl;
+    timers.Stop("lr_prediction");
+  }
```

Comment:
I realized while reviewing this today that this block should actually be inside the `if (params.Has("training"))` block above: if the user does not pass training data, but does pass `print_training_accuracy`, the option will not be ignored: we will try to print training accuracy anyway.

Reply:
I added the `if (params.Has("training"))` to the block, is that all?